packages/react/dialog/src/Dialog.tsx

```diff
@@ -221,15 +222,19 @@ const DialogContentModal = React.forwardRef((props, forwardedRef) => {
           // (closed !== unmounted when animating out)
           trapFocus={context.open}
           disableOutsidePointerEvents
+          onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+            event.preventDefault();
+            if (!isRightClickOutsideRef.current) context.triggerRef.current?.focus();
```

Comment:
Do we actually need this check here?

Because I see below that we say we don't close if it's a right click, therefore I'm thinking we never get to `onCloseAutoFocus` in the case of a right click?